cmakelib/QuickCppLibMakeDoxygen.cmake

```diff
@@ -1,6 +1,8 @@
 include(FindDoxygen)
 include(FindGit)
-if(NOT DOXYGEN_FOUND)
+if(${PROJECT_NAME}_DISABLE_DOC_GEN)
```

Comment:
Note that if doxygen is not found, the *target* for the docs generation is not added, and it is simply a target which will try to call doxygen so if someone tries that, they get failure. So this extra variable is superfluous.

Reply:
Yessir, I get that. But in the event I ever wanted to "build" QuickCppLib, I want to skip the doc step. Assumption being that Doxygen **is** installed, but I still don't want that step.

Reply:
Just because a target is created doesn't mean it is built! In fact, quickcpplib constructs a large graph of all possible build permutations, so many hundreds of targets per library. A reasonable default is set if the user doesn't ask for anything special, but it's a tiny subset of the targets available.

So for example `make afio_sl--testname` will make the test `testname` in the AFIO library with static linkage. `make afio_hl-asan-testname` will make the test `testname` in the AFIO library with header only linkage and address sanitiser enabled throughout everything.

Similarly, `make afio-docs` would generate doxygen based docs for the AFIO library. But it is never built by default. You have to ask for it. Does it make sense now why you don't need a cmake variable to enable or disable docs generation?

Reply:
@ned14 Yessir, I get that, re: targets exist as contrasted with targets built. Thanks for the clarification nonetheless.